### ISSUE TITLE
Staging - Character creation crashes server with 2 errors and caught signal 6

### ISSUE THREAD

**Issue by 5anitybane** [EvEmu-Project/evemu_Crucible#242]
**Describe the bug**
Trying to get set up for helping with dungeons.  Transition from character creation to in-game station crashes the server with 2x errors and caught signal 6.

**To Reproduce**
1. Create account through the initial log-in screen.
2. Create a Ni-Kunni male.
3. Select option 1 for top, bottom and feet.  No other selections.
4. Take character profile image as is - no pose or lighting changes etc.
5. Select Border Runners Ancestry.
6. Select Hedion University Education.
7. Error occurs almost half-way through the progress bar, just after it says 'Portrait Upload Successful.'

**Expected behavior**
Game loads into default station per character set-up with created character.

**Error**
- Inventory/InventoryItem.cpp:165 - invalid type returned for typeID 670.
- src/eve-core/memory/RefPtr.h:237: X* RefPtr<X>::operator->() const [with X = ShipItem]: Assertion '*this' failed.
- Caught signal 6.

**System Details (please complete the following information):**
 - Server OS: Ubuntu 20.04 Server
 - Direct build
 - EVEmu Version (latest staging from here; 0.8.4?)

**Additional context**
[eve-server.txt](https://github.com/EvEmu-Project/evemu_Crucible/files/10466840/eve-server.txt)

**Comment by jdhirst** [comment]
@Almamu I wonder if this is related to that other issue I ran into when making a new character the other day?

**Comment by Almamu** [comment]
I will have a look now, could be related, yes

**Comment by Almamu** [comment]
Your error seems to come from the Capsule creation for the player and not from the SelectCharacterID call in itself... Can you run this query against the database and send what the result is?:
```SELECT * FROM invTypes WHERE typeID = 670;```

It should return one record for the capsule, if It's not then there's something missing in the database (either the setup didn't complete or other issue), I guess you did the manual setup?

**Comment by Almamu** [comment]
Looks correct to me. Was there any other error in the log? Can you share the full log of this run? Looks like something didn't load properly on startup, but I cannot really say unless there's something in the log pointing to the error...

**Comment by 5anitybane** [comment]
I've attached the log to this reply.  It doesn't seem to point to a lot.
[21-01-2023-02-23.log](https://github.com/EvEmu-Project/evemu_Crucible/files/10468215/21-01-2023-02-23.log)

**Comment by Almamu** [comment]
I'm looking at this top to bottom but I don't really see why it would crash at that point :/ Have you tried it again since? I cannot reproduce the issue at all on my side either...

**Comment by 5anitybane** [comment]
Unfortunately it still occurs.  I'm just going to blast it and rebuild from scratch with new prerequisites in case something broke during build somehow, despite it succeeding.  I'll update once it's complete.

**Comment by 5anitybane** [comment]
Yep, rebuilt solution and it still occurs, I'm afraid.  I've put some messages into the server_error_discussion chat on Discord for now to see if there was something else wrong in prerequisite install.

**Comment by ZyenDomani** [comment]
your ItemFactory isnt creating the pod.
` const ItemType *iType = sItemFactory.GetType(data.typeID);
    if (iType == nullptr) {
        codelog(ITEM__ERROR, "II::CreateItemID() - Invalid type returned for typeID %u", data.typeID);
        return 0;
    }
`
dont know why, but if you wanna debug it, try enabling item logging and create new char.

i dont think this error will stop char creation, mostly because i dont remember if i put a test for that, but think i did.

either way, the item factory isnt creating the pod, and without tracing it, i dont know why.

Allan

**Comment by rokh85** [comment]
The same error is thrown with this log:

> EXCEPTION #6 logged at  05/19/2025 17:39:20 Unhandled exception in <TaskletExt object at 3ffc4870, abps=1001, ctxt='<NO CONTEXT>^<function <lambda> at (snip)>'>
> Caught at:
> /common/lib/bluepy.py(98) CallWrapper
> Thrown at:
> /common/lib/bluepy.py(86) CallWrapper
> /../carbon/client/script/ui/control/menu.py(517) <lambda>
> /client/script/ui/login/charsel.py(991) ChangeSlotOrder
> /client/script/ui/login/charsel.py(557) LoadCharacters
> /client/script/ui/login/charsel.py(627) AddCharacter
> /client/script/ui/login/charsel.py(578) GetDetails
>         self = form.CharSelection object at 0x8cb31b0, name=l_charsel, destroyed=False>
>         charID = 90000000
> TypeError: 'NoneType' object is not subscriptable
> Thread Locals:  session was <Session: (sid:2413247452195731585, clientID:0, mutating:0, corprole:0x0, userid:1, languageID:DE, role:0x2800000000000000L, address:127.0.0.1:49856, userType:30, sessionType:5, rolesAtAll:0x0, rolesAtHQ:0x0, rolesAtBase:0x0, rolesAtOther:0x0)>
> 
> EXCEPTION END

**Comment by ZyenDomani** [comment]
i am not familiar with the `ChangeSlotOrder` call, so not sure what's goin on there.
however, *something* is sending a PyNone in response to the character selection or loading calls, which is a problem.

i can no longer build the evemu server, so cant help debugging code.
do you have a log with this errror?

Allan

**Comment by rokh85** [comment]
Of course, here are the server log and the stack trace file of this occurance:

[evemu_client_stack_trace.txt](https://github.com/user-attachments/files/20326317/evemu_client_stack_trace.txt)
[19-05-2025-16-58.log](https://github.com/user-attachments/files/20326318/19-05-2025-16-58.log)

**Comment by ZyenDomani** [comment]
not logging enough data; hardly anything tbh
...well not logging the right data for this error, to be more precise.

when you get time, set the following in log.ini and send me the log, pls
CLIENT__CALL_REP=1
CLIENT__OUT_ALL=1
CLIENT__TRACE=1

warning...gonna be noisy, but shouldnt be too bad since the crash is early in connection.   since i dont know where the error could be (no basic calls shown in previous log), let's enable packet logging for everything.  this way i know we'll get the call im looking for.

is there a problem with your other account on your server?  i see you logged into acct 1 twice before making a new char on a diff account.
is that part of the error/crash, or you just wanted to make new char to show error?

Allan

**Comment by rokh85** [comment]
The issue occurs only after confirming character creation, so I had to make a new account to recreate the issue. As I have made another account to do it yet again now. Logging in after the crash with the same account will trigger an empty character selection screen where I cannot select anything (screenshot attached).

Scouring the database revealed to me, that the character were in fact created (if fully or in part I cannot tell), but are not transmitted on login correctly.

here the new log, trace and screenshot of the character selection screen:
[evemu_client_stack_trace.txt](https://github.com/user-attachments/files/20352317/evemu_client_stack_trace.txt)
[20-05-2025-18-47.log](https://github.com/user-attachments/files/20352316/20-05-2025-18-47.log)
![Image](https://github.com/user-attachments/assets/b1e66404-3286-409d-802e-33d2c743acf2)

**Comment by ZyenDomani** [comment]
well, shit.  nothing in the log or trace.  no clue why nothing is showing in the log...it should have at least 45 calls and full response dumps with those 3 flags turned on.

however, i had this same error a while back, after enabling *something* new  (been too long...dont remember what).
it is a latent bug that finally shown when something else was changed (fixed or modified, iirc).

anyway, i *THINK* its something to do with pod or ship, and is in the character init call, possibly from Client::Login() 
i seem to remember having to reorder the calls because something relied on something else being populated or some shit like that.

i only say this because it looks like staging branch is still (mostly) using my code, and quite probably has the same error.

do you know how to debug/step code?  if so, start in debug and find where segfault occurs, then restart and step thru that method.

if not, maybe @jdhirst or @Almamu can help with code stepping.

if needed i'll look thru my code and find the problem i had and solution i came up with....maybe duplicated here.

Allan

**Comment by jdhirst** [comment]
Apologies for being _that_ person but do you run into the same issue with docker?

**Comment by rokh85** [comment]
docker would not build at all (but it tried to install a whole lot of deps which were already present via dnf on my arch system, which naturally could not work). Because of this I did it manually via cmake. I can try again, but I do not have that much hope for another attempt for it to build now all of a sudden.

**Comment by ZyenDomani** [comment]
that log is from your client startup.
no server information there.

i cannot answer your question, but @jdhirst or @Almamu can once they see this.

**Comment by ZyenDomani** [comment]
nothing helpful there...just stops after the GetCharNewExtraCreationInfo() call.

*most* of the "no char on select" or "crash on create" is due to a bad method call/proc order on char login...see my post above from 20 May.

do you know how to debug/step code? if so, start in debug and find where segfault occurs, then restart and step thru that method.
as i neither use docker nor can build current evemu code, i cant fix this on official.

i have not had time to do anything with this in a while, and not sure when i'll be able to.
however, i'm fairly certain it is the call order causing a problem.
